File: Bai 01/AI Tools/prime_check.cpp
#include <iostream>
#include <vector>

using namespace std;

bool is_prime(int n) {
    if (n < 2) {
        return false;
    }

    for (int i = 2; i * i <= n; ++i) {
        if (n % i == 0) {
            return false;
        }
    }

    return true;
}

int main() {
    int n;
    cout << "Nhap n: ";
    cin >> n;

    vector<int> primes;
    for (int x = 2; x <= n; ++x) {
        if (is_prime(x)) {
            primes.push_back(x);
        }
    }

    cout << "Cac so nguyen to <= " << n << " la: ";
    cout << "[";
    for (size_t i = 0; i < primes.size(); ++i) {
        cout << primes[i];
        if (i + 1 < primes.size()) {
            cout << ", ";
        }
    }
    cout << "]" << endl;

    return 0;
}

File: Bai 01/AI Tools/prime_check.py
def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True

n = int(input("Nhập n: "))

primes = []
for x in range(2, n + 1):
    if is_prime(x):
        primes.append(x)

print("Các số nguyên tố <=", n, "là:", primes)
